### ISSUE TITLE
Version info is no longer printed to root.log on DaemonApp startup

### ISSUE THREAD

**Issue by joseph-wakeling-sociomantic** [sociomantic-tsunami/ocean#510]
Historically, application version info has always been printed to the root log (as well as the version log) on application startup, meaning that one always has a clear indication in the root log of when an application has freshly started.

As of the release of ocean v3.8.0/v4.1.0, this no longer happens.  `git bisect` reveals that it's down to commit https://github.com/sociomantic-tsunami/ocean/commit/c704d282612decfe485c7fd69c1f4bc13d90d6bd (PR https://github.com/sociomantic-tsunami/ocean/pull/490).  This looks like an unwanted side effect of an otherwise useful change.

**Comment by joseph-wakeling-sociomantic** [comment]
BTW for reference, here's the minimal app I used to track this down:

```D
module test_root_log.main;

import Version;

import ocean.transition;
import ocean.util.app.DaemonApp;

private class TestRootLogApp : DaemonApp
{
    import ocean.util.log.Logger;
    private Logger log;

    public this ()
    {
        super("test_root_log", "tests writing version info to root.log", version_info);
        this.log = Log.lookup("test_root_log");
    }

    protected override int run (Arguments args, ConfigParser config)
    {
        this.log.info("Hey, I'm running now");
        this.log.info("And now I'm shutting down");
        return 0;
    }
}

version (UnitTest) {} else
int main (istring[] cl_args)
{
    scope app = new TestRootLogApp;
    return app.main(cl_args);
}
```

```make
override LDFLAGS += -lebtree

export ASSERT_ON_STOMPING_PREVENTION=1

.PHONY: test_root_log
test_root_log: $B/test_root_log
$B/test_root_log: $C/src/test_root_log/main.d
all += test_root_log
```

and needed `.gitmodules` file:

```
[submodule "ocean"]
        path = submodules/ocean
        url = https://github.com/sociomantic-tsunami/ocean.git
[submodule "makd"]
        path = submodules/makd
        url = https://github.com/sociomantic-tsunami/makd.git
```

**Comment by nemanja-boric-sociomantic** [comment]
Fixed in #512.
